frameworklauncher/src/main/java/com/microsoft/frameworklauncher/common/model/TaskRoleStatus.java

```diff
@@ -18,11 +18,14 @@
 package com.microsoft.frameworklauncher.common.model;
 
 import java.io.Serializable;
+import java.util.List;
+import java.util.ArrayList;
 
 public class TaskRoleStatus implements Serializable {
   private String taskRoleName;
   private TaskRoleRolloutStatus taskRoleRolloutStatus;
   private Integer frameworkVersion;
+  private List<Range> portRanges = new ArrayList<>();
 
```

Comment:
TaskRole may not share the same portRanges. Please move it out here.

Reply:
BTW, even here, how did you ensure it is consistent with the allocated/requested? And how did you ensure the consistent when crash and recovery?